python/pyspark/sql/tests/test_utils.py

```diff
@@ -23,11 +23,100 @@
     IllegalArgumentException,
     SparkUpgradeException,
 )
+from pyspark.testing.utils import assertDFEquality
 from pyspark.testing.sqlutils import ReusedSQLTestCase
+from pyspark.testing.connectutils import ReusedConnectTestCase
+import pyspark.sql.functions as F
 from pyspark.sql.functions import to_date, unix_timestamp, from_unixtime
 
+import pandas as pd
```

Comment:
This import should be within each test, and guarded by `pyspark.testing.sqlutils.pandas_requirement_message`, e.g.,

```python
@unittest.skipIf(not have_pandas, pandas_requirement_message)  # type: ignore
```

Reply:
This is because pandas isn't a required but optional dependency in PySpark.